drivers/SmartThings/matter-rvc/src/init.lua

```diff
@@ -28,213 +25,232 @@ if version.api < 10 then
-  end
-  device:set_field(labels_field, labels_of_supported_modes, { persist = true })
+local function device_init(driver, device)
+  device:subscribe()
+  device:set_component_to_endpoint_fn(component_to_endpoint)
 end
 
-local function set_field_supported_modes_from_response(device, field_prefix, cluster_id, attribute_id, response)
-  for _, rb in ipairs(response.info_blocks) do
-    if rb.info_block.attribute_id == attribute_id and
-      rb.info_block.cluster_id == cluster_id and
-      rb.info_block.data.elements ~= nil then
-      set_field_supported_modes(device, field_prefix, rb.info_block.data.elements)
-      break
-    end
+local function do_configure(driver, device)
+  local clean_mode_eps = device:get_endpoints(clusters.RvcCleanMode.ID)
+  if #clean_mode_eps == 0 then
+    local profile_name = "rvc"
+    device.log.info_with({hub_logs=true}, string.format("Updating device profile to %s.", profile_name))
+    device:try_update_metadata({profile = profile_name})
   end
 end
 
-local function get_field_labels_of_supported_modes(device, field_prefix)
-  local field = string.format("%s_labels", field_prefix)
-  return device:get_field(field)
+local function info_changed(driver, device, event, args)
+  for cap_id, attributes in pairs(subscribed_attributes) do
+    if device:supports_capability_by_id(cap_id) then
+      for _, attr in ipairs(attributes) do
+        device:add_subscribed_attribute(attr)
+      end
+    end
+  end
+  device:subscribe()
 end
 
-local function get_labels_of_supported_modes(device, field_prefix, cluster_id, attribute_id, response)
-  local supported_modes = get_field_labels_of_supported_modes(device, field_prefix)
-  if supported_modes ~= nil then
-    return supported_modes
+-- Helper functions --
+local function update_supported_arguments(device, current_run_mode, operating_state)
+  device.log.info(string.format("update_supported_arguments: %s, %s", current_run_mode, operating_state))
+  if current_run_mode == nil or operating_state == nil then
+    return
   end
 
-  set_field_supported_modes_from_response(device,
-    field_prefix,
-    cluster_id,
-    attribute_id,
-    response)
-  return get_field_labels_of_supported_modes(device, field_prefix)
-end
-
-local function is_idle_mode(device, field_prefix, index, idle_mode_tag)
-  local mode_tags = get_field_mode_tags_of_supported_mode(device, field_prefix, index)
-  for _, mode_tag in ipairs(mode_tags) do
-    if mode_tag == idle_mode_tag then
-      return true
-    end
+  -- Error state
+  if operating_state == "Error" then
+    -- Set runMode to empty
+    local component = device.profile.components["runMode"]
+    local event = capabilities.mode.supportedArguments({}, {visibility = {displayed = false}})
+    device:emit_component_event(component, event)
+    -- Set cleanMode to empty
+    component = device.profile.components["cleanMode"]
+    device:emit_component_event(component, event)
```

Comment:
```suggestion
    local empty_event = capabilities.mode.supportedArguments({}, {visibility = {displayed = false}})
    -- Set runMode to empty
   device:emit_component_event(device.profile.components["runMode"], event)
    -- Set cleanMode to empty
   device:emit_component_event(device.profile.components["cleanMode"], event)
```
maybe we could write it more like this? Sorry, the original wording confused me a lot when I first read it.

Reply:
I will modify it~!